app/controllers/foreman_inventory_upload/uploads_controller.rb

```diff
@@ -14,14 +14,7 @@ def download_file
       files = Dir["{#{ForemanInventoryUpload.uploads_file_path(filename)},#{ForemanInventoryUpload.done_file_path(filename)}}"]
 
       return send_file files.first, disposition: 'attachment', filename: filename unless files.empty?
-      throw_flash_error "File doesn't exist"
-    end
-
-    def throw_flash_error(message)
-      process_error(
-        :redirect => foreman_inventory_upload_index_path,
-        :error_msg => message
-      )
+      raise ::Foreman::Exception.new("The report file doesn't exist")
```

Comment:
We should probably use `not_found` method from [`ApplicationController`](https://github.com/theforeman/foreman/blob/develop/app/controllers/application_controller.rb#L105) with the exception:

```suggestion
      return not_found ::Foreman::Exception.new("The report file doesn't exist")
```

Reply:
`return not_found` ends up with a page like this one, looks like the warning is based on the `controller`:

![Selection_001](https://user-images.githubusercontent.com/26363699/107875574-7aa44580-6ec9-11eb-97a5-a36114059b84.png)

I think the `::Foreman::Exception` is better, have more info in it:

![Screenshot (58)](https://user-images.githubusercontent.com/26363699/107875619-afb09800-6ec9-11eb-86ee-c148ae157688.png)